tests/core/base/_utils/test_format_template.py

```diff
@@ -22,6 +22,7 @@ def test_format_template(
     mock_get_template_values.return_value = attrs
     template = """
     Recommend a {genre} book.
+    
```

Comment:
What is this line for?

Reply:
the ensure that we strip extra whitespace to reduce tokens.